src/process.rs

```diff
@@ -15,37 +15,28 @@ pub struct Line {
 pub type Graph = Vec<Line>;
 pub type Graphs = HashMap<String, Graph>;
 
-fn fsts<K, V>(hm: &HashMap<K, V>) -> Result<Vec<f64>>
-where
-    K: NumCast + Ord + Copy + Debug,
-{
-    hm.iter()
-        .sorted_by_key(|x| x.0)
-        .map(|x| <f64 as NumCast>::from(*x.0).ok_or(anyhow!("Failed casting {:?} to f64", x.0)))
-        .collect()
-}
-fn snds<K, V>(hm: &HashMap<K, V>) -> Result<Vec<f64>>
-where
-    K: Ord,
-    V: NumCast + Copy + Debug,
-{
-    hm.iter()
-        .sorted_by_key(|x| x.0)
-        .map(|x| <f64 as NumCast>::from(*x.1).ok_or(anyhow!("Failed casting {:?} to f64", x.1)))
-        .collect()
-}
-
 impl Line {
-    pub fn try_new<S, K, V>(label: S, hm: &HashMap<K, V>) -> Result<Line>
+    pub fn try_new<K, V>(label: impl Into<String>, hm: &HashMap<K, V>) -> Result<Line>
```

Comment:
We can also drop the specificity on `HashMap` by allowing to pass any container that could iterate over its data in coordinate like tuples `(K, V)`


```suggestion
    pub fn try_new<K, V>(label: impl Into<String>, hm: impl IntoIterator<Item = (K, V)>) -> Result<Line>
```

With this we'd need to change some of the callers to pass an owned data though